src/_P103_Atlas_EZO_pH_ORP_EC.ino

```diff
@@ -34,496 +36,528 @@ boolean Plugin_103(byte function, struct EventStruct *event, String& string)
+          break;
+        case '1':
+          addFormNote(F("<span style='color:green'>Single point calibration ok</span>"));
+          break;
+        case '2':
+          addFormNote(F("<span style='color:green'>Two points calibration ok</span>"));
+          break;
         }
-        break;
       }
+    }
+    break;
+    }
 
-      // Temperature compensation
-      if(board_type == PH || board_type == EC) {
-        double value;
-        char strValue[5];
-
-        addFormSubHeader(F("Temperature compensation"));
-        char deviceTemperatureTemplate[40];
-        LoadCustomTaskSettings(event->TaskIndex, (byte *)&deviceTemperatureTemplate, sizeof(deviceTemperatureTemplate));
-            addFormTextBox(F("Temperature "), F("Plugin_103_temperature_template"), deviceTemperatureTemplate, sizeof(deviceTemperatureTemplate));
-            addFormNote(F(
-                            "You can use a formulae and idealy refer to a temp sensor (directly, via ESPEasyP2P or MQTT import) ,e.g. '[Pool#Temperature]'. If you don't have a sensor, you could type a fixed value like '25' for '25.5'."));
-
-            String deviceTemperatureTemplateString(deviceTemperatureTemplate);
-            String pooltempString(parseTemplate(deviceTemperatureTemplateString, 40));
-
-            if (Calculate(pooltempString.c_str(), value) != CalculateReturnCode::OK) {
-                addFormNote(F("It seems I can't parse your formulae. Fixed value will be used!"));
-                value = FIXED_TEMP_VALUE;
-            }
-
-            dtostrf(value, 5, 2, strValue);
-            String actualValueStr(F("Actual value: "));
-            actualValueStr += strValue;
-            addFormNote(actualValueStr);
-        }
+    // Temperature compensation
+    if (board_type == PH || board_type == EC)
+    {
+      double value;
+      char strValue[5];
 
-      success = true;
-      break;
+      addFormSubHeader(F("Temperature compensation"));
+      char deviceTemperatureTemplate[40] = {0};
+      LoadCustomTaskSettings(event->TaskIndex, (byte *)&deviceTemperatureTemplate, sizeof(deviceTemperatureTemplate));
+      addFormTextBox(F("Temperature "), F("Plugin_103_temperature_template"), deviceTemperatureTemplate, sizeof(deviceTemperatureTemplate));
+      addFormNote(F(
+          "You can use a formulae and idealy refer to a temp sensor (directly, via ESPEasyP2P or MQTT import) ,e.g. '[Pool#Temperature]'. If you don't have a sensor, you could type a fixed value like '25' for '25.5'."));
+
+      String deviceTemperatureTemplateString(deviceTemperatureTemplate);
+      String pooltempString(parseTemplate(deviceTemperatureTemplateString, 40));
+
+      if (Calculate(pooltempString.c_str(), value) != CalculateReturnCode::OK)
+      {
+        addFormNote(F("It seems I can't parse your formulae. Fixed value will be used!"));
+        value = FIXED_TEMP_VALUE;
+      }
+
+      dtostrf(value, 5, 2, strValue);
```

Comment:
This will fill the entire array of `strValue` so no zero termination guaranteed.
It probably has a zero at the next address as it has a length which is not a multiple of 4, but it is not guaranteed to be unused.
So better set the length slightly longer and zero terminate it yourself.

Reply:
fixed